plugin/vimux.vim

```diff
@@ -3,20 +3,24 @@ if exists('g:loaded_vimux') || &compatible
 endif
 let g:loaded_vimux = 1
 
-function! s:VimuxOption(option, default)
-  if exists(a:option)
-    return eval(a:option)
-  else
-    return a:default
-  endif
-endfunction
-
-function! s:VimuxTmuxCmd()
-  return s:VimuxOption('g:VimuxTmuxCommand', 'tmux')
-endfunction
-
-if !executable(s:VimuxTmuxCmd())
-  echohl ErrorMsg | echomsg 'Failed to find executable '.s:VimuxTmuxCmd() | echohl None
+" Set up all global options with defaults right away, in one place
+let g:VimuxDebug         = get(g:, 'VimuxDebug',         v:false)
+let g:VimuxHeight        = get(g:, 'VimuxHeight',        20)
+let g:VimuxOpenExtraArgs = get(g:, 'VimuxOpenExtraArgs', '')
+let g:VimuxOrientation   = get(g:, 'VimuxOrientation',   'v')
+let g:VimuxPromptString  = get(g:, 'VimuxPromptString',  'Command? ')
+let g:VimuxResetSequence = get(g:, 'VimuxResetSequence', 'q C-u')
+let g:VimuxRunnerName    = get(g:, 'VimuxRunnerName',    '')
+let g:VimuxRunnerType    = get(g:, 'VimuxRunnerType',    'pane')
+let g:VimuxTmuxCommand   = get(g:, 'VimuxTmuxCommand',   'tmux')
+let g:VimuxUseNearest    = get(g:, 'VimuxUseNearest',    v:true)
+
+function! s:VimuxOption(name) abort
```

Comment:
I still think we need to do something about the removal of the `eval()` here. I don't have any way of knowing how many people *use* this function. I suppose we need to look through all the add-on plugins out there.

I wonder if adding a global default setting for whether to treat all other settings as static values or code to eval would do the trick. Or is there no hard to just using eval all the time?

Reply:
I've reconsidered this. Looking through the code history I think this got introduced as a bad port from Ruby to VimL and perhaps even the author didn't understand what was going on.

I think we can move forward without re-implementing this. If it turns out I am wrong about that and we break something, here is the patch we'll need to fix it:

```patch
From f514091113e2dce147c5b20fe82d23fa7edc1ad2 Mon Sep 17 00:00:00 2001
From: Caleb Maclennan <caleb@alerque.com>
Date: Tue, 16 Feb 2021 16:02:57 +0300
Subject: [PATCH] Revert regression, resume evaluating setting vars

Signed-off-by: Caleb Maclennan <caleb@alerque.com>
---
 plugin/vimux.vim | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/plugin/vimux.vim b/plugin/vimux.vim
index 1cf6d3c..69a2141 100644
--- a/plugin/vimux.vim
+++ b/plugin/vimux.vim
@@ -16,7 +16,12 @@ let g:VimuxTmuxCommand   = get(g:, 'VimuxTmuxCommand',   'tmux')
 let g:VimuxUseNearest    = get(g:, 'VimuxUseNearest',    v:true)
 
 function! VimuxOption(name) abort
-  return get(b:, a:name, get(g:, a:name))
+  let l:value = get(b:, a:name, get(g:, a:name))
+  if exists(l:value)
+    return eval(l:value)
+  else
+    return l:value
+  endif
 endfunction
 
 if !executable(VimuxOption('VimuxTmuxCommand'))
-- 
2.30.1

```